tmt/steps/execute/internal.py

```diff
@@ -150,6 +170,43 @@ def check(self, test):
         else:
             return self.check_shell(test)
 
+    def _setup_reboot(self, guest):
+        """ Prepare the guest environment for potential reboot """
+        # We only want to change the environment in tmt-provisioned machines,
+        # ignore Local and Connect
+        if isinstance(guest, GuestLocal) or guest.__class__ is Guest:
+            return
```

Comment:
Hmmm, we need to think this through... I can imagine that for `connect` it could be useful to support reboot as well...

Reply:
Hm, I mentioned above that this is a leftover but I see one problem with `connect`. If we want to support reboot on a guest, we must always replace the `rstrnt-reboot` implementation even if no actual reboot is needed. So this would modify the connected system every time which may be problematic for some users...

Reply:
Yes I see your concern. Don't see a clear solution here. Some brainstorm: What about enabling the reboot support upon request?

```yaml
execute:
    how: tmt
    reboot: true
```

Or something like this:

```yaml
execute:
    how: tmt
    features: reboot
```

This would prevent doing reboot preparations when they are not needed. But would require explicit settings in the plan.

Reply:
The first option sounds great to me. This way nothing with tmt changes when using "old" plans but if you learn about the reboot feature and want to enable it for a new plan/test, you can easily do so. I can try to implement it as a draft for now but lets discuss the details on Tuesday at tmt hacking to see what others think

Reply:
> lets discuss the details on Tuesday at tmt hacking to see what others think

Agreed. Added to the agenda.